diff -druN ../score/DelegationManager.sol core/DelegationManager.sol
--- ../score/DelegationManager.sol	2023-10-02 14:44:51
+++ core/DelegationManager.sol	2023-10-02 15:05:39
@@ -351,9 +351,11 @@
      */
     function decreaseDelegatedShares(
         address staker,
-        IStrategy[] calldata strategies,
-        uint256[] calldata shares
-    ) external onlyStrategyManagerOrEigenPodManager {
+        // MUNGED calldata => memory
+        IStrategy[] memory strategies,
+        uint256[] memory shares
+        // MUNGED external => public
+    ) public onlyStrategyManagerOrEigenPodManager {
         // if the staker is delegated to an operator, then subtract strategy shares from the operator's shares
         if (isDelegated(staker)) {
             address operator = delegatedTo[staker];
