diff -druN ../score/DelegationManager.sol core/DelegationManager.sol
--- ../score/DelegationManager.sol	2023-09-19 13:13:31
+++ core/DelegationManager.sol	2023-09-19 13:25:35
@@ -265,8 +265,11 @@
      * @dev *If the staker is actively delegated*, then decreases the `staker`'s delegated shares in each entry of `strategies` by its respective `shares[i]`. Otherwise does nothing.
      * @dev Callable only by the StrategyManager or EigenPodManager.
      */
-    function decreaseDelegatedShares(address staker, IStrategy[] calldata strategies, uint256[] calldata shares, bool undelegateIfPossible)
-        external
+    function decreaseDelegatedShares(
+        // MUNGED calldata => memory
+        address staker, IStrategy[] memory strategies, uint256[] memory shares, bool undelegateIfPossible)
+        // MUNGED external => public
+        public
         onlyStrategyManagerOrEigenPodManager
     {
         // if the staker is delegated to an operator
