diff -druN ../score/DelegationManager.sol core/DelegationManager.sol
--- ../score/DelegationManager.sol	2023-01-13 14:12:34
+++ core/DelegationManager.sol	2023-01-13 14:24:43
@@ -160,10 +160,10 @@
      */
     function decreaseDelegatedShares(
         address staker,
-        IStrategy[] calldata strategies,
-        uint256[] calldata shares
+        IStrategy[] memory strategies, // MUNGED calldata => memory
+        uint256[] memory shares // MUNGED calldata => memory
     )
-        external
+        public // MUNGED external => public
         onlyStrategyManager
     {
         if (isDelegated(staker)) {