diff -druN ../score/DelegationManager.sol core/DelegationManager.sol
--- ../score/DelegationManager.sol	2023-09-26 13:12:35
+++ core/DelegationManager.sol	2023-09-26 13:40:04
@@ -307,10 +307,12 @@
      * @dev Callable only by the StrategyManager or EigenPodManager.
      */
     function decreaseDelegatedShares(
-        address staker,
-        IStrategy[] calldata strategies,
-        uint256[] calldata shares
-    ) external onlyStrategyManagerOrEigenPodManager {
+        address staker, 
+        // MUNGED calldata => memory
+        IStrategy[] memory strategies,
+        uint256[] memory shares
+        // MUNGED external => public
+    ) public onlyStrategyManagerOrEigenPodManager {
         // if the staker is delegated to an operator
         if (isDelegated(staker)) {
             address operator = delegatedTo[staker];
